src/cleanfigure.m

```diff
@@ -554,18 +554,28 @@ function simplifyLine(meta, handle, targetResolution)
     % The resolution is lost only beyond the multiplier magnification
     mult = 2;
 
-    % Convert data to pixel units, magnify and mark only the first
-    % point that occupies a given position
-    mask = [true; diff(round(x * xToPix * mult))~=0];
-    mask = [true; diff(round(y * yToPix * mult))~=0] | mask;
+    % Convert data to pixel units and magnify
+    dataPixel = round([x * xToPix * mult, ...
+                       y * yToPix * mult]);
 
-    % Keep end points or it might truncate whole pixels
+    % Sort the pixels
+    [dataPixelSorted, id_orig] = sortrows(dataPixel);
+
+    % Find the duplicate pixels
+    mask_sorted = [true; diff(dataPixelSorted(:,1))~=0 | ...
+                         diff(dataPixelSorted(:,2))~=0];
+
+    % Unwind the sorting
+    mask            = false(size(x));
+    mask(id_orig)   = mask_sorted;
+
+    % Set the first, last, as well as unique pixels to true
+    mask(1)         = true;
+    mask(end)       = true;
+
+    % Set NaNs to true
     inan         = isnan(x) | isnan(y);
-    df           = diff([false; inan; false]);
-    istart       = df == 1;
-    pend         = find(df == -1)-1;
-    mask(istart) = true;
-    mask(pend)   = true;
```

Comment:
@okomarov thx ;)

I think the easy solution is to pass
```
xData - xLim(1)
yData - yLim(1)
```
This should ensure, that everything is in the correct pixel, as it eliminates negative values, which could possibly come from different rounding for negative/positive values.